### DISCUSSION TITLE
Export to image in Figma seems broken under stealth mode

### DISCUSSION THREAD

**Discussion by GrassBlock1** [kkapsner/CanvasBlocker#726] [Q&A] [answered]
When I try to export a frame in figma when canvasblocker is on with stealth mode, the export preview is empty and a transparent image is downloaded:
![image](https://github.com/user-attachments/assets/7e9c8241-388b-4faa-8015-bce2f94b28f8)
However when canvasblocker is off, this works normally again:
![image](https://github.com/user-attachments/assets/7eac2008-4436-4423-96a7-814dcea7bde5)

How can I fix this? Any help will be appreciated.

**Comment by spodermenpls** [comment, marked as answer]
And what happens, when you use the "convenient" preset instead of the "stealth" preset?

**Reply by GrassBlock1** [reply]
> And what happens, when you use the "convenient" preset instead of the "stealth" preset?

Well, it works again when I set this preset and restart the browser.
![image](https://github.com/user-attachments/assets/d0081260-13aa-44a0-8142-45cac0b6f09a)

And I switched back again, it seems work too now.
But I discovered when I open Developer Tools like picture shown before, it's still broken on both modes:
![image](https://github.com/user-attachments/assets/c7634a1c-41d7-44a0-a543-eba8a224502e)

**Reply by spodermenpls** [reply]
You are giving conflicting assessments, "it works again" and "it seems to work now too", but also "it's broken on both modes".. is the developer console necessary to determine if Figma works or not? I would imagine that one would be able to tell just by using the website. What you can do besides switching between presets, is switching off API protections one by one (inside CanvasBlocker's settings page), to see if one of them is at fault, and if so, you can whitelist `figma.com` inside the site-specific settings of that API protection. Otherwise you can also switch to the latest beta version of CanvasBlocker (version 1.11, to be found here: https://canvasblocker.kkapsner.de/versions/?C=M;O=D), which possibly could help with your issue.

**Reply by GrassBlock1** [reply]
> You are giving conflicting assessments, "it works again" and "it seems to work now too", but also "it's broken on both modes".. is the developer console necessary to determine if Figma works or not? I would imagine that one would be able to tell just by using the website. What you can do besides switching between presets, is switching off API protections one by one (inside CanvasBlocker's settings page), to see if one of them is at fault, and if so, you can whitelist `figma.com` inside the site-specific settings of that API protection. Otherwise you can also switch to the latest beta version of CanvasBlocker (version 1.11, to be found here: https://canvasblocker.kkapsner.de/versions/?C=M;O=D), which possibly could help with your issue.

Sorry for my unclear clarification about my issue.

> it works again.

the "it" here stands for exporting under convenice mode.

>  it seems work too now.

the "it" here stands for exporting under stealth mode.

>  it's still broken on both modes:

this is what I see when dev tools siderbar is open.

Anyway,  I think opening a developer tools is unnecessary to use figma, so this condition(opening the devtools) may be a rare case.

And I tried the beta version of canvasblocker. When it runs under stealth mode, the result was the same with the stable version(broken exports). And under convenience mode, the export will be work again. (I didn't open devtools to test these though)

edit: it seems that under stealth mode export would be broken randomly...The error wasn't be 100% reproducible. I'll fully investigate this later.
